JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -371,27 +359,57 @@ private fun TabWithTopics(
  */
 @Composable
 private fun TabWithSections(
-    sections: TopicsMap,
+    sections: List<InterestSection>,
     selectedTopics: Set<TopicSelection>,
     onTopicSelect: (TopicSelection) -> Unit
 ) {
-    LazyColumn(Modifier.navigationBarsPadding()) {
-        sections.forEach { (section, topics) ->
-            item {
-                Text(
-                    text = section,
-                    modifier = Modifier
-                        .padding(16.dp)
-                        .semantics { heading() },
-                    style = MaterialTheme.typography.subtitle1
-                )
+    BoxWithConstraints {
+        val windowSize = remember(maxWidth) { getWindowSize(maxWidth) }
+        val columns = remember(windowSize) { if (windowSize == WindowSize.Compact) 1 else 2 }
+
+        val itemMaxWidth = rememberItemMaxWidth(windowMaxWidth = maxWidth, columns = columns)
+        // Group interest sections given the number of columns that fill the screen.
+        // Sections are distributed evenly on the screen using the modulus operator.
+        val groupedSections: Array<MutableList<InterestSection>> = remember(columns) {
+            val sectionsInColumns = Array<MutableList<InterestSection>>(columns) { mutableListOf() }
+            sections.forEachIndexed { index, section ->
+                val column = index % columns
+                sectionsInColumns[column] += section
             }
-            items(topics) { topic ->
-                TopicItem(
-                    itemTitle = topic,
-                    selected = selectedTopics.contains(TopicSelection(section, topic))
-                ) { onTopicSelect(TopicSelection(section, topic)) }
-                TopicDivider()
+            sectionsInColumns
+        }
+
+        Row(
+            horizontalArrangement = Arrangement.SpaceEvenly,
+            modifier = Modifier.navigationBarsPadding().fillMaxSize()
+        ) {
+            groupedSections.forEach { sectionsInGroup ->
+                LazyColumn(
+                    Modifier
+                        .widthIn(max = itemMaxWidth)
+                        .padding(horizontal = if (columns > 1) 8.dp else 0.dp)
+                ) {
+                    sectionsInGroup.forEach { (section, topics) ->
```

Comment:
One thing that's a little strange with the current logic: the number of items in each column isn't really balanced. If all of the even sections have 20 items and all of the odd sections have 2 items, then they could be really unbalanced, which is a bit strange.

Reply:
Sure, fixed this. As the algorithm is more expensive if we also try to find the shortest column, I moved the execution to the background thread 👍